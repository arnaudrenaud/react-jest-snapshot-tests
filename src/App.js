import React, { Component } from 'react';
import './App.css';

class App extends Component {
  state = {
    advice: '',
  }

  async componentDidMount() {
    this.loadAdvice();
  }

  loadAdvice = async () => {
    try {
      const response = await fetch('https://api.adviceslip.com/advice');
      const { slip: { advice } } = await response.json();
      this.setState({ advice });
    } catch(error) {
      this.setState({advice: ''});
    }
  }

  render() {
    return (
      <div className="App">
        <p className="Advice">{this.state.advice || 'â€¦'}</p>
        <button
          className="Advice-other-advice-button"
          onClick={this.loadAdvice}
        >
          Show other advice
        </button>
      </div>
    );
  }
}

export default App;
