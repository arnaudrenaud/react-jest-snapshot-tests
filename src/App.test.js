import React from 'react';
import { shallow } from 'enzyme';
import fetchMock from 'fetch-mock';

import App from './App';

describe('App', () => {
  beforeEach(() => {
    fetchMock.get(
      'https://api.adviceslip.com/advice',
      { slip : { advice: 'Always trust random advice.'} },
      { repeat: 1 }
    );
  });

  afterEach(() => {
    fetchMock.reset();
  });

  describe('before advice fetched', () => {
    it('renders correctly', () => {
      const wrapper = shallow(<App />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('after advice fetched', () => {
    it('renders correctly', done => {
      const wrapper = shallow(<App />);
      setImmediate(() => {
        expect(wrapper).toMatchSnapshot();
        done();
      });
    });

    describe('when button clicked', () => {
      beforeEach(() => {
        fetchMock.get(
          'https://api.adviceslip.com/advice',
          { slip : { advice: 'Other advice.'} },
          { overwriteRoutes: false }
        );
      });

      it('renders other random advice', done => {
        const wrapper = shallow(<App />);
        wrapper.find('.Advice-other-advice-button').simulate('click');
        setImmediate(() => {
          expect(wrapper).toMatchSnapshot();
          done();
        });
      });
    });
  });
});
