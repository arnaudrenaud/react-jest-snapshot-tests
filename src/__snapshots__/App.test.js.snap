// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App after advice fetched renders correctly 1`] = `
<div
  className="App"
>
  <p
    className="Advice"
  >
    Always trust random advice.
  </p>
  <button
    className="Advice-other-advice-button"
    onClick={[Function]}
  >
    Show other advice
  </button>
</div>
`;

exports[`App after advice fetched when button clicked renders other random advice 1`] = `
<div
  className="App"
>
  <p
    className="Advice"
  >
    Other advice.
  </p>
  <button
    className="Advice-other-advice-button"
    onClick={[Function]}
  >
    Show other advice
  </button>
</div>
`;

exports[`App before advice fetched renders correctly 1`] = `
<div
  className="App"
>
  <p
    className="Advice"
  >
    â€¦
  </p>
  <button
    className="Advice-other-advice-button"
    onClick={[Function]}
  >
    Show other advice
  </button>
</div>
`;
